src/hotspot/cpu/ppc/assembler_ppc.hpp

```diff
@@ -1424,6 +1507,28 @@ class Assembler : public AbstractAssembler {
   static bool is_lis(int x) {
      return is_addis(x) && inv_ra_field(x)==0;
   }
+  static bool is_paddi_prefix(int x) {
+     return PADDI_PREFIX_OPCODE == (x & PADDI_PREFIX_OPCODE_MASK);
+  }
+  static bool is_paddi_suffix(int x) {
+     return PADDI_SUFFIX_OPCODE == (x & PADDI_SUFFIX_OPCODE_MASK);
+  }
+  // This function can skip a nop for 64-byte alignment and check the next word for a prefix.
+  // Since the alignement nop is uncommon case, we will keep callers of this function simple,
+  // as they which are often assertions and complex if-statement.
+  static int is_paddi(const int* p, bool is_pli = false) {
+     int32_t* p_inst = (int32_t*)p;
+
+     if (is_aligned(reinterpret_cast<uintptr_t>(p_inst+1), 64) && is_nop(*p_inst)) {
+        p_inst++;  // skip over nop
+     }
+     if (is_paddi_prefix(p_inst[0]) && is_paddi_suffix(p_inst[1])) {
+        return !is_pli ||
+               (inv_r_eo(p_inst[0]) == 0 && inv_ra_field(p_inst[1]) == 0);
+     } else {
+        return false;
+     }
+  }
```

Comment:
I can't see any usage of it. As Corey stated, unused code should better get removed because it doesn't get tested and may break unless you're planning to use it in the near future or need it for debugging purposes.